### ISSUE TITLE
Proposal: Dart updater/installer

### ISSUE THREAD

**Issue by NexisHunter** [dart-lang/sdk#39894]
### Feature

Flutter has the ability to be upgraded via the command line. I would like to see something like this for Dart. This also may be able to offer an option to act as an installer for more Linux distributions aside from Debian based distributions by providing a dart2native compiled executable that can download and install the SDK, and update it if there is a new version, with the option to change the "channel" of the SDK. I am willing to assist and develop the feature, I'm just unsure where to start ing the SDK housing the files.

### System Information
Dart VM version: 2.7.0-dev.2.1 
Operating System: Linux
Arch: x64

**Comment by devoncarew** [comment]
Thanks for the feature request (and the offer for contributions)! cc'ing @jwren and @mit-mit for consideration.

**Comment by NexisHunter** [comment]
Sounds good and I look forward to helping :)

**Comment by NexisHunter** [comment]
Hey quick update I built a possible POC, the repo is [here](github.com/nexishunter/dnfi).

**Comment by mit-mit** [comment]
Thanks for reaching out @NexisHunter!

We're considering reworking the overall Dart CLI toolchain to be more similar to the Flutter cli; I'd like to tackle update as part of that work, and not as a separate workstream.

**Comment by NexisHunter** [comment]
@mit-mit I completely agree that was just a pic I had been working on to suit the use I had for the other day, if you'd like some assistance left me know. I'd love to help when and where possible.

**Comment by mit-mit** [comment]
Thanks, let's keep this open until we have something more tangible I can point you at

**Comment by NexisHunter** [comment]
Sounds great!

**Comment by mit-mit** [comment]
Unassigning self. We now have a `dart` command in place, and this could be extended with a new `upgrade` command. This command should likely be based on the archives available here: https://dart.dev/get-dart

As for logic to determine the latest SDKs available, perhaps the logic here is instructional: https://github.com/dart-lang/setup-dart/blob/main/setup.sh
